README.md

```diff
@@ -0,0 +1,376 @@
+		clientId, _ := client.CreateClientID(context.Background())
+		client.Subscribe(context.TODO(), clientId, testRoomId)
+		clientIds = append(clientIds, clientId)
+	}
+
+	ticker := time.NewTicker(time.Second)
+	done := false
+	for !done {
+		select {
+		case <-doneC:
+			done = true
+		case <-ticker.C:
+			for _, clientId := range clientIds {
+				// Must call periodically for each Client ID & Room ID comibnation to keep
+				// Client ID alive.
+				client.Ping(context.Background(), clientId)
+			}
+
+			metrics, _ := client.GetMetrics(context.TODO(), &redisSyncFanoutQueue.GetApiMetricsOptions{
+				TopRoomsLimit: 10,
+			})
+
+			fmt.Printf("Metrics: %v\n", metrics)
+		}
+	}
+	ticker.Stop()
+
+	for _, clientId := range clientIds {
+		client.Unsubscribe(context.Background(), clientId, testRoomId)
+	}
+
+	client.Close()
+}
+
+func main() {
+	sub(1)
+}
+```
+
+### Delivering and ACKnowledging messages (worker)
+
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+
+	redisSyncFanoutQueue "github.com/zavitax/redis-sync-fanout-queue-go"
+)
+
+var testMessageContent = "test message content"
+var testRoomId = "GO-ROOM-TEST"
+
+var redisOptions = &redis.Options{
+	Addr:     "127.0.0.1:6379",
+	Password: "",
+	DB:       0,
+}
+
+func createApiOptions() *redisSyncFanoutQueue.ApiOptions {
+	result := &redisSyncFanoutQueue.ApiOptions{
+		RedisOptions:   redisOptions,
+		ClientTimeout:  time.Second * 15,
+		RedisKeyPrefix: fmt.Sprintf("{test-redis-sync-fanout-queue}::%v", "test"),
+	}
+
+	return result
+}
+
+func createWorkerOptions() *redisSyncFanoutQueue.WorkerOptions {
+	result := &redisSyncFanoutQueue.WorkerOptions{
+		RedisOptions:   redisOptions,
+		RedisKeyPrefix: fmt.Sprintf("{test-redis-sync-fanout-queue}::%v", "test"),
+	}
+
+	return result
+}
+
+func createApiClient(options *redisSyncFanoutQueue.ApiOptions) (redisSyncFanoutQueue.RedisQueueApiClient, error) {
+	return redisSyncFanoutQueue.NewApiClient(context.TODO(), options)
+}
+
+func worker() {
+	doneC := make(chan os.Signal, 1)
+	signal.Notify(doneC)
+
+	// Used for ACKs
+	client, _ := createApiClient(createApiOptions())
+
+	wo := createWorkerOptions()
+	wo.HandleRoomClientTimeout = func(ctx context.Context, clientId *string, roomId *string) error {
+		return nil
+	}
+	wo.HandleMessage = func(ctx context.Context, clientId *string, msg *redisSyncFanoutQueue.Message) error {
+		if msg.Data == nil {
+			fmt.Printf("Received nil data\n")
+			return nil
+		}
+
+		strData := (*msg.Data).(string)
+		if strData != testMessageContent {
+			fmt.Printf("Expected '%v' but received '%v'\n", testMessageContent, strData)
+			return nil
+		}
+
+		fmt.Printf("Received: client[%s] room[%s]: %v\n", *clientId, msg.Room, strData)
+
+		if msg.AckToken != nil {
+			//time.Sleep(time.Second * 3)
+			// This should be called by the actual client, when it's done processing the message
+			client.AckMessage(ctx, *clientId, msg.AckToken)
```

Comment:
I'm considering those separate entities since ActToken _may_ be packaging additional information moving forward (for example: a digital signature and an expiration date/time come to mind). The relation of client ID to an AckToken is internal and as far as the interface consumer is concerned - are completely different pieces of data.